Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+// ===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// ===----------------------------------------------------------------------===//
+
+import Dispatch
+import Foundation
+import PackagePlugin
+
+@main
+struct AWSLambdaPackager: CommandPlugin {
+    func performCommand(context: PackagePlugin.PluginContext, arguments: [String]) async throws {
+
+        let configuration = try Configuration(context: context, arguments: arguments)
+        guard !configuration.products.isEmpty else {
+            throw Errors.unknownProduct("no appropriate products found to deploy")
+        }
+
+        let samExecutable = try context.tool(named: "sam")
+        let samExecutableUrl = URL(fileURLWithPath: samExecutable.path.string)
+        let deploymentDescriptorExecutableUrl = configuration.deployExecutable
```

Comment:
do you think there is an opportunity to do this via configuration instead of code? if so, we have another type of plugin that helps with source generation. 

if we stay with the approach that the user must create a module named in specific way, we should probably add validation step to check if the executable (deploymentDescriptorExecutableUrl) exists on disk

Reply:
Agree.  I will propose a simplified approach where function developer must just provide `Deploy.swift` file, similar to SPM's `Package.swift`.

The plugin will compile and run this file to generate the SAM deployment descriptor.

That will remove the need for the function developer to provide an executable target.  The declarative nature of Deploy.swift will be more familiar and less error prone.